### PR TITLE
Lights

### PR DESCRIPTION
**Pull request by patriciogonzalezvivo**

### CODE REVIEW

File: core/src/scene/light.h [hunk 1]
```diff
@@ -4,20 +4,21 @@
...
     DIRECTIONAL,
     POINT,
     SPOT,
     CUSTOM
 };
 
+enum class LightOrigin {
+    CAMERA,
+    GROUND,
+    WORLD
```
Comment (blair1618) [inline core/src/scene/light.h]: What would the difference between "ground" and "world" origins be?

Comment (bcamper) [inline core/src/scene/light.h]: `ground` is like a point dropped from the current camera location on the ground:

https://github.com/tangrams/tangram/wiki/Lights-Overview#origin

Perhaps it should be a cast ray intersecting the ground plane in this case? On web, it is (currently) always looking straight down...

Comment (patriciogonzalezvivo) [inline core/src/scene/light.h]: GROUND is the camera position (in XY) projected on the floor with Z is on the ground
WORLD is absolute world position

More info at https://github.com/tangrams/tangram/wiki/Lights-Overview#origin

Comment (blair1618) [inline core/src/scene/light.h]: I see, thanks. For the first pass we can take the easy route and drop straight down from the camera. We should experiment with projecting from the center from the screen onto the ground too, it's not obvious to me a priori that one is always better.

Comment (patriciogonzalezvivo) [inline core/src/scene/light.h]: https://github.com/tangrams/tangram-es/blob/lights_origin/core/src/scene/pointLight.cpp#L54

**Comment by blair1618** [comment]
Looks good to me. @tallytalwar you had a chance to look over this?

**Pull request merged**